vyper/codegen/expr.py

```diff
@@ -206,11 +206,12 @@ def parse_Name(self):
     def parse_Attribute(self):
         typ = self.expr._metadata["type"]
 
-        # MyFlag.foo
-        if (
-            isinstance(typ, FlagT)
-            and isinstance(self.expr.value, vy_ast.Name)
-            and typ.name == self.expr.value.id
+        # (lib1).MyFlag.foo
+        if isinstance(typ, FlagT) and (
+                (isinstance(self.expr.value, vy_ast.Name)
+                 and typ.name == self.expr.value.id)
+                or (isinstance(self.expr.value, vy_ast.Attribute)
```

Comment:
i don't think this works for things like `lib1.lib2.Flag.FOO`

Reply:
we could probably drop the AST checks entirely once we know that it's an instance of FlagT -- `if isinstance(typ, FlagT): ... <flagname = self.expr.attr>`

Reply:
fixed in https://github.com/vyperlang/vyper/pull/3871/commits/6dfd08d57e07b23929706bfa4aae7b466271841b